src/main/scala/firrtl/ir/IR.scala

```diff
@@ -22,24 +24,73 @@ case object NoInfo extends Info {
   override def toString: String = ""
   def ++(that: Info): Info = that
 }
-case class FileInfo(info: StringLit) extends Info {
-  override def toString: String = " @[" + info.serialize + "]"
+
+/** Stores the string of a file info annotation in its escaped form. */
+case class FileInfo(escaped: String) extends Info {
```

Comment:
Unfortunately this is a backwards breaking change and there's no way to provide the old `unapply` and deprecate it. You're _supposed_ to be able to, but there's a bug in Scala scala/bug 11252 (not linking because I don't want to put them on blast).

I think this is worth the improvement and the best we can do we I'll accept it but I won't like it 🙅 .

Reply:
> I think this is worth the improvement and the best we can do we I'll accept it but I won't like it

Yeah. I think at this point it is either breaking backwards compatibility or we decide to abandon the PR.